src/configuration.cpp

```diff
@@ -193,12 +193,14 @@ ssr::conf_struct ssr::configuration(int& argc, char* argv[])
   // load system-wide config file (Linux et al.)
   load_config_file("/etc/ssr.conf",conf);
   // load user config file (Mac)
-  std::string filename = getenv("HOME");
+  std::string filename = pathtools::get_home_dir().string();
```

Comment:
Wouldn't it make more sense to keep using a `path` here?
Then the following line could use `/` to append components.

Reply:
Unfortunately the cross-compiler doesn't like these implicit path to string conversions...

Reply:
Sorry, I wasn't clear.

I'm not talking about any implicit conversions (those are bad!), I'm talking about never using `std::string` at all.

`get_home_dir()` would return a `path`, which would then be manipulated as such, and in the very end, `.c_str()` would be called on it.

Reply:
Oh, that's unfortunate. It's interesting how C++ reliably manages to have the least expected behavior by default ... but I guess `string().c_str()` is fine.